corehq/apps/app_manager/models.py

```diff
@@ -2173,21 +2176,44 @@ class CaseSearchProperty(DocumentSchema):
     Case properties available to search on.
     """
     name = StringProperty()
-    label = DictProperty()
+    label = LabelProperty()
     appearance = StringProperty(exclude_if_none=True)
     input_ = StringProperty(exclude_if_none=True)
     default_value = StringProperty(exclude_if_none=True)
-    hint = DictProperty()
+    hint = LabelProperty()
     hidden = BooleanProperty(default=False)
     allow_blank_value = BooleanProperty(default=False)
     exclude = BooleanProperty(default=False)
-    required = StringProperty(exclude_if_none=True)
-    validation = SchemaListProperty(CaseSearchValidationCondition)
+    required = SchemaProperty(Assertion)
+    validations = SchemaListProperty(Assertion)
 
     # applicable when appearance is a receiver
     receiver_expression = StringProperty(exclude_if_none=True)
     itemset = SchemaProperty(Itemset)
 
+    @classmethod
+    def wrap(cls, data):
+        original = deepcopy(data)
+        required = data.pop('required', None)
+        if required:
+            # TODO its always str
+            if isinstance(required, str):
+                data['required'] = {'test': required}
+            elif isinstance(required, list):
+                data['required'] = required[0]
+            else:
+                data['required'] = required
+
+        old_validations = data.pop('validation', None)  # it was changed to plural
```

Comment:
Wasn't `validation` added in https://github.com/dimagi/commcare-hq/pull/31871/ which isn't merged yet? Is this to fix apps on staging?

Reply:
yeah, in case QA has already started, I wanted to try not to break it.